crates/sui/src/ethereum_client_commands.rs

```diff
@@ -232,3 +258,359 @@ pub(crate) async fn init_ethereum_state(
+/// # Errors
+/// The function returns an error if any of the following occur:
+/// - Missing Ethereum state configuration.
+/// - Errors during object deserialization.
+/// - Failure in fetching updates or proofs from the Ethereum light client.
+/// - Transaction construction or execution failures.
+pub(crate) async fn eth_approve_message(
+    context: &mut WalletContext,
+    eth_dwallet_cap_id: ObjectID,
+    message: String,
+    dwallet_id: ObjectID,
+    gas: Option<ObjectID>,
+    gas_budget: u64,
+    serialize_unsigned_transaction: bool,
+    serialize_signed_transaction: bool,
+) -> Result<SuiClientCommandResult, anyhow::Error> {
+    let active_env = context.config.get_active_env()?.clone();
+    let mut eth_lc_config = get_eth_config(context)?;
+
+    let latest_state_object_id = match active_env.eth_client_settings {
+        Some(eth_client_settings) => eth_client_settings
+            .state_object_id
+            .ok_or_else(|| anyhow!("ETH State object ID configuration not found"))?,
+        None => return Err(anyhow!("ETH State object ID configuration not found")),
+    };
+
+    let eth_dwallet_cap_data_bcs = get_object_bcs_by_id(context, eth_dwallet_cap_id).await?;
+    let eth_dwallet_cap_obj: EthDWalletCap = eth_dwallet_cap_data_bcs
+        .try_as_move()
+        .ok_or_else(|| anyhow!("Object is not a Move Object"))?
+        .deserialize()
+        .map_err(|e| anyhow!("Error deserializing object: {e}"))?;
+
+    let latest_eth_state_data_bcs = get_object_bcs_by_id(context, latest_state_object_id).await?;
+    let latest_eth_state_obj: LatestEthStateObject = latest_eth_state_data_bcs
+        .try_as_move()
+        .ok_or_else(|| anyhow!("Object is not a Move Object"))?
+        .deserialize()
+        .map_err(|e| anyhow!("Error deserializing object: {e}"))?;
+
+    let eth_state_object_id = latest_eth_state_obj.eth_state_id;
+    let eth_state_data_bcs = get_object_bcs_by_id(context, eth_state_object_id).await?;
+    let eth_state_obj: EthStateObject = eth_state_data_bcs
+        .try_as_move()
+        .ok_or_else(|| anyhow!("Object is not a Move Object"))?
+        .deserialize()
+        .map_err(|e| anyhow!("Error deserializing object: {e}"))?;
+
+    let mut eth_state = bcs::from_bytes::<ConsensusStateManager<NimbusRpc>>(&eth_state_obj.data)
+        .map_err(|e| anyhow!("error parsing eth state data: {e}"))?;
+    let mut eth_state = eth_state.set_rpc(&eth_lc_config.consensus_rpc);
+
+    let latest_eth_state_shared_object =
+        get_shared_object_input_by_id(context, latest_state_object_id).await?;
+
+    if let Some(checkpoint) = eth_state.last_checkpoint.clone() {
+        eth_lc_config.checkpoint = format!("0x{}", encode(checkpoint));
+    } else {
+        return Err(anyhow!("Checkpoint is missing in the Ethereum state"));
+    }
+
+    let data_slot = eth_dwallet_cap_obj.eth_smart_contract_slot;
+    let contract_addr: String = eth_dwallet_cap_obj.eth_smart_contract_addr;
```

Comment:
Since we changed the design to have just a single smart contract per network (and not per dwallet), this should be changed (here and in the network code)

Reply:
Done